src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_pix2pix_zero.py

```diff
@@ -265,9 +293,36 @@ def enable_sequential_cpu_offload(self, gpu_id=0):
         for cpu_offloaded_model in [self.unet, self.text_encoder, self.vae]:
             cpu_offload(cpu_offloaded_model, device)
 
+        if self.caption_generator is not None:
+            cpu_offload(self.caption_generator, execution_device=device)
+
         if self.safety_checker is not None:
             cpu_offload(self.safety_checker, execution_device=device, offload_buffers=True)
 
+    def enable_model_cpu_offload(self, gpu_id=0):
+        r"""
+        Offloads all models to CPU using accelerate, reducing memory usage with a low impact on performance. Compared
+        to `enable_sequential_cpu_offload`, this method moves one whole model at a time to the GPU when its `forward`
+        method is called, and the model remains in GPU until the next model runs. Memory savings are lower than with
+        `enable_sequential_cpu_offload`, but performance is much better due to the iterative execution of the `unet`.
+        """
+        if is_accelerate_available() and is_accelerate_version(">=", "0.17.0.dev0"):
+            from accelerate import cpu_offload_with_hook
+        else:
+            raise ImportError("`enable_model_offload` requires `accelerate v0.17.0` or higher.")
+
+        device = torch.device(f"cuda:{gpu_id}")
+
+        hook = None
+        for cpu_offloaded_model in [self.vae, self.text_encoder, self.unet, self.vae]:
```

Comment:
Repetition in the `self.vae`. You probably meant `self.captioner`?

Reply:
That's actually on purpose so that the first self.vae is offloaded when the text encoder is called :sweat_smile:

Reply:
cc @pcuenca this should work no? Since inversion is img2img

Reply:
Could you elaborate this a bit? Didn't really understand it.

Reply:
Hmm, no worries. I will sync up with you offline on this next week.